### ISSUE TITLE
Frequent disconnects on otherwise perfectly stable Internet connection

### ISSUE THREAD

**Issue by jdolan** [psychs/limechat#295]
I've been using LimeChat for a few years, and I love it. Great piece of software. For whatever reason, tho, it often has a difficult time maintaining connection to all IRC networks. X-Chat does not have this problem. Other apps that rely on a persistent connection (Slack, Mail.app, etc..) also have far fewer issues throughout the course of the day.

Here's a screenshot of what I'm talking about:
![screen shot 2016-09-05 at 1 14 48 pm](https://cloud.githubusercontent.com/assets/643118/18254397/4a529610-736b-11e6-92ec-c9cb12377ae3.png)

I see this from my home (CenturyLink DSL), at work (Comcast Business cable), and on most other networks I frequent. To limit the noise this causes on public channels, I've been using a `znc` bouncer that I run on `quetoo.org`, which is a Rackspace server that I have a 60ms ping and zero packet loss to from my house. LimeChat behaves the same as if it were connecting to the IRC servers directly: constant, regular disconnects.

Any ideas or workarounds that I can try?

Thanks,

jdolan

**Comment by jdolan** [comment]
A bit more detail, here's a portion of the log for one of the disconnects on Freenode:

```
12:34 jdolan has changed mode: +i
12:34 Reply(306): You have been marked as being away
12:34 Reply(305): You are no longer marked as being away
12:34 NickServ: You are already logged in as jdolan.
12:54 Operation timed out
12:54 Disconnected
12:55 Reconnecting…
12:56 Operation timed out
12:56 Disconnected
12:56 Reconnecting…
12:57 Operation timed out
12:57 Disconnected
12:58 Reconnecting…
12:59 Operation timed out
12:59 Disconnected
12:59 Reconnecting…
13:01 Operation timed out
13:01 Disconnected
13:01 Reconnecting…
13:01 Connected
13:01 Logged in
```

**Comment by jdolan** [comment]
Just to further illustrate the point, today I was on a video call for 90 minutes, which went completely uninterrupted. 720p video with 44KHz audio, for 90 minutes. No drops. Meanwhile, LimeChat disconnected roughly a dozen times during this same interval. I can't be the only user experiencing this, can I?

**Comment by jdolan** [comment]
Well kiss my grits. I rebooted the machine (I'll typically go weeks or months without doing so), and the problem has resolved itself. LimeChat (which I had restarted numerous times prior to rebooting) went from disconnecting every 3-10 minutes for weeks on end, to remaining connected for 6 hours and counting. This wasn't the first time I had seen this behavior from LimeChat, so I don't think it's a fluke. But at least now I know that a reboot will fix the problem
